sys/net/link_layer/ieee802154/ieee802154.c

```diff
@@ -27,6 +27,14 @@ size_t ieee802154_set_frame_hdr(uint8_t *buf, const uint8_t *src, size_t src_len
     uint8_t type = (flags & IEEE802154_FCF_TYPE_MASK);
     uint8_t bcast = (flags & IEEE802154_BCAST);
 
+    if (dst_pan.u16 == src_pan.u16 &&
+        (src_len != 0 && dst_len != 0)) {
```

Comment:
the problem with this comparison of `dst_len != 0` is that for multicast messages such as RS the length actually is 0 see [1]. The result is the for RS messages we have SRC_PAN==DST_PAN but `dst_len=0` and the will not be unset which makes Linux drops those RS and never send an RA.

This could be fixed, for instance in [2], by not not passing `netif_hdr->dst_l2addr_len`directly but rather set this like with `src_len`. That is, if the packet is broadcast or multicast (see [3]) then set `dst_len = 2`. 



[1]: https://github.com/RIOT-OS/RIOT/blob/master/sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c#L588

[2]: https://github.com/RIOT-OS/RIOT/blob/master/sys/net/gnrc/link_layer/netdev2/gnrc_netdev2_ieee802154.c#L185

[3]: